src/contracts/pods/EigenPodManager.sol

```diff
@@ -407,94 +236,73 @@ contract EigenPodManager is
+        uint256 currentShareDeficit = podOwnerShareDeficit[podOwner];
+
+        // skip dealing with deficit if there isn't any
+        if (currentShareDeficit == 0) {
+            podOwnerShares[podOwner] += shareAmount;
+            return shareAmount;            
+        }
 
-        // if the podOwner has delegated shares, record an increase in their delegated shares
-        if (!isInUndelegationLimbo(podOwner)) {
-            delegationManager.increaseDelegatedShares(podOwner, beaconChainETHStrategy, shareAmount);
+        // get rid of the whole deficit if possible
+        if (shareAmount >= currentShareDeficit) {
+            podOwnerShareDeficit[podOwner] = 0;
+            shareAmount -= currentShareDeficit;
+            podOwnerShares[podOwner] += shareAmount;
+            return shareAmount;
+        // otherwise get rid of as much deficit as possible
+        } else {
+            podOwnerShareDeficit[podOwner] -= shareAmount;
+            return 0;
         }
+
     }
 
-    // @notice Reduces the `podOwner`'s shares by `shareAmount` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
-    function _removeShares(address podOwner, uint256 shareAmount) internal {
+    // @notice Reduces the `podOwner`'s shares by `shareAmount`, adding to deficit if necessary
+    // @dev Returns the number of shares removed from `podOwnerShares[podOwner]`
+    function _removeShares(address podOwner, uint256 shareAmount) internal returns (uint256) {
         require(podOwner != address(0), "EigenPodManager._removeShares: depositor cannot be zero address");
-        require(shareAmount != 0, "EigenPodManager._removeShares: shareAmount should not be zero!");
 
         uint256 currentPodOwnerShares = podOwnerShares[podOwner];
-        require(shareAmount <= currentPodOwnerShares, "EigenPodManager._removeShares: shareAmount too high");
-
-        unchecked {
-            currentPodOwnerShares = currentPodOwnerShares - shareAmount;
-        }
 
-        podOwnerShares[podOwner] = currentPodOwnerShares;
-
-        // if the podOwner has delegated shares, record a decrease in their delegated shares
-        if (!isInUndelegationLimbo(podOwner)) {
-            uint256[] memory shareAmounts = new uint256[](1);
-            shareAmounts[0] = shareAmount;
-            IStrategy[] memory strategies = new IStrategy[](1);
-            strategies[0] = beaconChainETHStrategy;
-            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts);
+        // skip dealing with deficit if there isn't any need for it
+        if (shareAmount <= currentPodOwnerShares) {
+            podOwnerShares[podOwner] = currentPodOwnerShares - shareAmount;
+            return shareAmount;
+        // otherwise, add to the deficit as necessary
+        } else {
+            podOwnerShares[podOwner] = 0;
+            uint256 newDeficitAmount = (shareAmount - currentPodOwnerShares);
```

Comment:
parentheses only if inlining below?

Reply:
ehhh I'm fine leaving it

Reply:
addressed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/227 (merged)